Pegasus/include/pegasus/Math.hpp

```diff
@@ -942,6 +942,30 @@ void CalculateDotProductForeach(
 }
 
 /**
+ * @brief Calculates arbitrary orthonormal vector to the given one
+ * @tparam VectorType glm vector of size 3
+ * @param vector of interest
+ * @return orthonormal vector to the vector of interest
+ */
+template < typename VectorType >
+VectorType CalculateOrthogonalVector(VectorType vector)
+{
+    VectorType result;
+
+    for (uint8_t i = 0; i < result.length(); ++i)
+    {
+        if (vector[i] != 0.0)
+        {
+            result[(1 + i) % 3] = vector[i];
+            result[i] = -vector[(1 + i) % 3];
+        }
+    }
+    result = glm::normalize(result);
+
+    return result;
```

Comment:
return glm::normalize(result);?

Reply:
It will be NRVO'ed anyway, what's the difference?

Reply:
This just a taste question. For me 1 line better then 3, but its not critical, so you decide